1-WebApp-OIDC/1-7-B2C-Embedded/README.md

```diff
@@ -0,0 +1,247 @@
+
+## Option 1 - Run the pre-configured sample
+
+1. Build the solution and run it.
+1. Open your web browser and make a request to the app. Accept the IIS Express SSL certificate if needed. Click on SignIn/Up button.
+1. Click on Sign-In
+
+## Option 2 (Recommended) - Configure the sample with your own B2C app
+
+### Step 2: Get your own Azure AD B2C tenant
+
+If you don't have an Azure AD B2C tenant yet, you'll need to create an Azure AD B2C tenant by following the [Tutorial: Create an Azure Active Directory B2C tenant](https://azure.microsoft.com/documentation/articles/active-directory-b2c-get-started).
+
+### Step 3: Enable custom domains for Azure Active Directory B2C
+
+To use Azure AD B2C Embedded sign-in, you need to first enable a custom domain with your Azure AD B2C account, and configure it with a custom domain which is on the same domain as your published application, i.e. **Same Origin**. This is required because Azure AD B2C should not allow you to load the sign-in experience in an iframe unless you enable CORS for the source domains loading the iframe using a custom policy. So first, enable a custom domain similar to **login.yourcustomdomain.com** if your application is published on something similar to **www.yourcustomdomain.com**. [Tutorial: Enable custom domains for Azure Active Directory B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/custom-domain?pivots=b2c-custom-policy).
+
+>Note: This will effectively disallow you from using sign-in if the application is run from a development machine. To get around that, you may want to create a separate Azure AD Tenant for development, with a development version of the custom policy which has CORS enabled for **<https://localhost:XXXX>**. Also note that this may result in an undesirable user experience Because Azure AD B2C session cookies within an iframe are considered third-party cookies, certain browsers (for example Safari or Chrome in incognito mode) either block or clear these cookies. This should never be done in your production Azure AD B2C tenant.
+
+### Step 4: Create your SignUp/SignIn, Password Reset, and Profile Edit policies (custom policy)
+
+This sample depends on a custom sign-up/sign-in policy. This is because you need to customize the policy to allow **JourneyFraming**, where you allow the user journey to be loaded into an iframe.
+You can use one of the policies in the [Azure AD B2C Custom Policy Starter Pack](https://github.com/Azure-Samples/active-directory-b2c-custom-policy-starterpack). Follow the [Tutorial: Create user flows in Azure Active Directory B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-create-user-flows?pivots=b2c-custom-policy) to deploy such policy.
+
+In your policy, aside from the normal steps in customizing the policy, make sure to add the following in **SignUpOrSignIn.xml**, **PasswordReset.xml**, and **ProfileEdit.xml**, inside the `<RelyingParty>` tag, right after the `<DefaultUserJourney>` tag.
```

Comment:
@helshabini would it be possible to include these policies (or examples of them) in the sample? in a folder maybe?

Reply:
I'm submitted another pull request (still pending) which includes these policies in a different repository. But I guess including them here again wouldn't hurt.